assembly/src/ast/parsers/io_ops.rs

```diff
@@ -336,8 +347,22 @@ fn parse_hex_value(op: &Token, param_str: &str, param_idx: usize) -> Result<u64,
         ));
     }
 
-    let value = u64::from_str_radix(param_str, 16)
-        .map_err(|_| ParsingError::invalid_param(op, param_idx))?;
+    let value = match endianness {
+        Endianness::Big => u64::from_str_radix(param_str, 16)
+            .map_err(|_| ParsingError::invalid_param(op, param_idx))?,
+        Endianness::Litte => {
+            if param_str.len() != 16 {
+                return Err(ParsingError::invalid_param_with_reason(
+                    op,
+                    param_idx,
+                    &format!("hex string '{param_str}' being a part of long hex string should contain 16 characters"),
+                ));
+            }
+            u64::from_str_radix(param_str, 16)
+                .map(|v| v.swap_bytes())
+                .map_err(|_| ParsingError::invalid_param(op, param_idx))?
+        }
+    };
```

Comment:
I would probably first check endianness and only then check for even number of chars etc. as for long hex string, if the check for 64 chars passes, we don't need to check anything else. So, it would look something like this:
```Rust
let value = match endianness {
    Endianness::Big => {
        // check if hex string length is divisible by 2
        // check if hex string length is smaller than or equal to HEX_CHUNK_SIZE
        // parse hex string
    }
    Endianness::Litte => {
        // check that hex string length is equal to HEX_CHUNK_SIZE
        // parse hex string
    }
};
```